src/metrics/filters/filter_benchmark_test.go

```diff
@@ -125,6 +125,14 @@ func BenchmarkRangeFilterRangeMatchNegation(b *testing.B) {
 	benchRangeFilterRange(b, []byte("!a-z"), []byte("p"), false)
 }
 
+func BenchmarkMultiRangeFilterPrefixCollision(b *testing.B) {
```

Comment:
@prateek PTAL

Reply:
To Justin's point - please add a comment to the code with "After change results"

And in the PR, update description with both before/after results

Reply:
Updated the PR description to add benchmark results and a table to help with comparison. Also updated the code to add a small optimization to stop iterating through patterns if an optimal match was found and updated tests to ensure it was working

Reply:
I ran it with `b.ReportAllocs()` and there are 0 allocs with the old and new logic. 

Updated the tests to show the change for all benchmark tests related to my code. There is a small difference (~0-15%) in test cases where the prefix/suffix collision does not occur likely due to the additional conditionals present after a prefix or suffix detection is found.